packages/stack/proxy/src/rpc-manager.ts

```diff
@@ -0,0 +1,196 @@
+    this.events.setCommandHandler("rpc:response:interceptor:register", this.registerResponseInterceptor.bind(this));
+  }
+
+  private async onProxyRequest<T>(params: ProxyRequestParams<T>, callback: Callback<ProxyRequestParams<T>>) {
+    try {
+      params = await this.executeInterceptors(this.requestInterceptors, params);
+    } catch (err) {
+      err.message = __("Error executing RPC request modifications for '%s': %s", params?.request?.method, err.message);
+      return callback(err);
+    }
+    callback(null, params);
+  }
+
+  private async onProxyResponse<T, R>(params: ProxyResponseParams<T, R>, callback: Callback<ProxyResponseParams<T, R>>) {
+    try {
+      params = await this.executeInterceptors(this.responseInterceptors, params);
+    } catch (err) {
+      err.message = __("Error executing RPC response modifications for '%s': %s", params?.request?.method, err.message);
+      return callback(err);
+    }
+    callback(null, params);
+  }
+
+  private registerRequestInterceptor<T>(filter: string | RegExp, interceptor: RpcRequestInterceptor<T>, options: RegistrationOptions = { priority: 50 }, callback?: Callback<null>) {
+    this.requestInterceptors.push({ filter, options, interceptor });
+    if (callback) {
+      callback();
+    }
+  }
+
+  private registerResponseInterceptor<T, R>(filter: string | RegExp, interceptor: RpcResponseInterceptor<T, R>, options: RegistrationOptions = { priority: 50 }, callback?: Callback<null>) {
+    this.responseInterceptors.push({ filter, options, interceptor });
+    if (callback) {
+      callback();
+    }
+  }
+
+  private async resetAccounts(cb: Callback<null>) {
+    this._nodeAccounts = null;
+    this._accounts = null;
+    cb();
+  }
+
+  private async executeInterceptors<T>(
+    registrations: Array<RpcRequestRegistration<T>>,
+    params: ProxyRequestParams<T>
+  ): Promise<ProxyRequestParams<T>>;
+  private async executeInterceptors<T, R>(
+    registrations: Array<RpcResponseRegistration<T, R>>,
+    params: ProxyResponseParams<T, R>
+  ): Promise<ProxyResponseParams<T, R>>;
+  private async executeInterceptors<T, R>(
+    registrations: Array<RpcRequestRegistration<T> | RpcResponseRegistration<T, R>>,
+    params: ProxyRequestParams<T> | ProxyResponseParams<T, R>
+  ): Promise<ProxyRequestParams<T> | ProxyResponseParams<T, R>> {
+    const web3 = await this.web3;
+    const accounts = await this.accounts;
+    const nodeAccounts = await this.nodeAccounts;
+    const { method } = params.request;
+    const registrationsToRun = registrations
+      .filter(registration => this.shouldModify(registration.filter, method))
+      .sort((a, b) => this.sortByPriority(a.options.priority, b.options.priority));
+
+    for (const registration of registrationsToRun) {
+      params = await registration.interceptor(params);
+    }
+    return params;
+  }
+
+  private shouldModify(filter: string | string[] | RegExp, method: string) {
```

Comment:
Prolly better to rename to `shouldIntercept()`

Reply:
Yeah